pkg/machines/scripts/create_machine.sh

```diff
@@ -53,7 +53,7 @@ if [ "$SOURCE_TYPE" = "pxe" ]; then
 elif [ "$START_VM" = "true" ]; then
     if [ "$SOURCE_TYPE" = "disk_image" ]; then
         INSTALL_METHOD="--import"
-    elif [ "${SOURCE#/}" != "$SOURCE" ] && [ -f "${SOURCE}" ]; then
+    elif ( [ "${SOURCE#/}" != "$SOURCE" ] && [ -f "${SOURCE}" ] ) || ( [ "$SOURCE_TYPE" = "url" ] && [ ${SOURCE: -4} = ".iso" ] ); then
```

Comment:
`${SOURCE: -4}` is a bashism, it doesn't work in e. g. dash:
```
# SOURCE="foo.iso"; echo ${SOURCE: -4}
dash: 1: Bad substitution
```
And the script is /bin/sh. Also, this needs quoting. This can be done in a POSIX shell way with `[ "${SOURCE%.iso}" != "$SOURCE" ]`.

Reply:
Addressed your comment.